uds/database/data_record/abstract_data_record.py

```diff
@@ -50,7 +51,8 @@
 """
 
 
-class DecodedDataRecord(TypedDict):
+@dataclass
```

Comment:
Nice, I have not seen `dataclass` package before but it is exactly what w need here.